tools/lxdclient/client.go

```diff
@@ -229,39 +245,60 @@ func newRawClient(remote Remote) (*lxd.Client, error) {
 		}
 	}
 
-	/* If this is the LXD provider on the localhost, let's do an extra
-	 * check to make sure that lxdbr0 is configured.
-	 */
-	if remote.ID() == remoteIDForLocal {
-		profile, err := client.ProfileConfig("default")
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
+	return client, nil
+}
 
-		/* If the default profile doesn't have eth0 in it, then the
-		 * user has messed with it, so let's just use whatever they set up.
-		 *
-		 * Otherwise, if it looks like it's pointing at our lxdbr0,
-		 * let's check and make sure that is configured.
-		 */
-		eth0, ok := profile.Devices["eth0"]
-		if ok && eth0["type"] == "nic" && eth0["nictype"] == "bridged" && eth0["parent"] == "lxdbr0" {
-			conf, err := ioutil.ReadFile(LXDBridgeFile)
-			if err != nil {
-				if !os.IsNotExist(err) {
-					return nil, errors.Trace(err)
-				} else {
-					return nil, bridgeConfigError("lxdbr0 configured but no config file found at " + LXDBridgeFile)
-				}
-			}
+// verifyDefaultProfileBridgeConfig takes a LXD API client and extracts the
+// network bridge configured on the "default" profile. Additionally, if the
+// default "lxdbr0" bridge is used, its configuration in LXDBridgeFile is also
+// inspected to make sure it has a chance to work.
+func verifyDefaultProfileBridgeConfig(client *lxd.Client) (string, error) {
+	const (
+		defaultProfileName = "default"
+		configTypeKey      = "type"
+		configTypeNic      = "nic"
+		configNicTypeKey   = "nictype"
+		configBridged      = "bridged"
+		configEth0         = "eth0"
+		configParentKey    = "parent"
+	)
+
+	config, err := client.ProfileConfig(defaultProfileName)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
 
-			if err = checkLXDBridgeConfiguration(string(conf)); err != nil {
-				return nil, errors.Trace(err)
-			}
+	// If the default profile doesn't have eth0 in it, then the
+	// user has messed with it, so let's just use whatever they set up.
+	//
+	// Otherwise, if it looks like it's pointing at bridge, and it is the
+	// default "lxdbr0", verify that is configured correctly.
+	eth0, ok := config.Devices[configEth0]
+	if !ok || eth0[configTypeKey] != configTypeNic || eth0[configNicTypeKey] != configBridged {
+		return "", errors.Errorf("unexpected LXD %q profile config: %+v", defaultProfileName, config)
+	}
+
+	bridgeName := eth0[configParentKey]
+	logger.Infof(`LXD "default" profile uses network bridge %q`, bridgeName)
+
+	if bridgeName != network.DefaultLXDBridge {
+		return bridgeName, nil
+	}
+
+	bridgeConfig, err := ioutil.ReadFile(LXDBridgeFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
```

Comment:
Flip the sense of this and switch the legs of the if around - double negatives are confusing! 

Actually, `os.IsNotExist` can take nil, so it might be clearer to get rid of the nesting:
```
if os.IsNotExist(err) {
    return "", bridgeConfigError("lxdbr0 configured but no config file found at " + LXDBridgeFile)
else if err != nil {
    return "", errors.Trace(err)
}
```